docs/template_plugin/tests/functional/op_reference/ctc_greedy_decoder.cpp

```diff
@@ -0,0 +1,132 @@
+        function = std::make_shared<ov::Function>(NodeVector{decoder}, ParameterVector{data, indices});
+        return function;
+    }
+};
+
+TEST_P(ReferenceCTCGreedyDecoderTest, CompareWithRefs) {
+    Exec();
+}
+
+template <element::Type_t IN_ET>
+std::vector<CTCGreedyDecoderParams> generateParams() {
+    using T = typename element_type_traits<IN_ET>::value_type;
+    std::vector<CTCGreedyDecoderParams> params {
+        CTCGreedyDecoderParams(
+            Tensor(IN_ET, {3, 1, 3}, std::vector<T>{0.1f, 0.2f, 0.f, 0.4f, 0.3f, 0.f, 0.5f, 0.6f, 0.f}),
+            Tensor(IN_ET, {3, 1}, std::vector<T>{1.0f, 1.0f, 1.0f}),
+            false,
+            Tensor(IN_ET, {1, 3, 1, 1}, std::vector<T>{1.0f, 0.0f, 1.0f}),
+            "ctc_greedy_decoder"),
+        CTCGreedyDecoderParams(
+            Tensor(IN_ET, {3, 2, 3}, std::vector<T>{
+                0.1f, 0.2f, 0.f, 0.15f, 0.25f, 0.f, 0.4f, 0.3f, 0.f, 0.45f, 0.35f, 0.f, 0.5f, 0.6f, 0.f, 0.55f, 0.65f, 0.f}),
+            Tensor(IN_ET, {3, 2}, std::vector<T>{1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f}),
+            false,
+            Tensor(IN_ET, {2, 3, 1, 1}, std::vector<T>{1.0f, 0.0f, 1.0f, 1.0f, 0.0f, 1.0f}),
+            "ctc_greedy_decoder_multiple_batches"),
+        CTCGreedyDecoderParams(
+            Tensor(IN_ET, {3, 1, 3}, std::vector<T>{0.1f, 0.2f, 0.f, 0.4f, 0.3f, 0.f, 0.5f, 0.6f, 0.f}),
+            Tensor(IN_ET, {3, 1}, std::vector<T>{1.0f, 1.0f, 0.0f}),
+            false,
+            Tensor(IN_ET, {1, 3, 1, 1}, std::vector<T>{1.0f, 0.0f, -1.0f}),
+            "ctc_greedy_decoder_single_batch_short_sequence"),
+        CTCGreedyDecoderParams(
+            Tensor(IN_ET, {3, 1, 3}, std::vector<T>{0.1f, 0.2f, 0.f, 0.3f, 0.4f, 0.f, 0.6f, 0.5f, 0.f}),
+            Tensor(IN_ET, {3, 1}, std::vector<T>{1.0f, 1.0f, 1.0f}),
+            true,
+            Tensor(IN_ET, {1, 3, 1, 1}, std::vector<T>{1.0f, 0.0f, -1.0f}),
+            "ctc_greedy_decoder_merge"),
+        CTCGreedyDecoderParams(
+            Tensor(IN_ET, {3, 1, 3}, std::vector<T>{0.1f, 0.2f, 0.f, 0.3f, 0.4f, 0.f, 0.6f, 0.5f, 0.f}),
+            Tensor(IN_ET, {3, 1}, std::vector<T>{1.0f, 1.0f, 1.0f}),
+            false,
+            Tensor(IN_ET, {1, 3, 1, 1}, std::vector<T>{1.0f, 1.0f, 0.0f}),
+            "ctc_greedy_decoder_single_no_merge"),
+        CTCGreedyDecoderParams(
+            Tensor(IN_ET, {2, 2, 3}, std::vector<T>{0.1f, 0.2f, 0.f, 0.4f, 0.3f, 0.f, 0.5f, 0.6f, 0.f, 0.7f, 0.8f, 0.f}),
+            Tensor(IN_ET, {2, 2}, std::vector<T>{1.0f, 1.0f, 1.0f, 0.0f}),
+            false,
+            Tensor(IN_ET, {2, 2, 1, 1}, std::vector<T>{1.0f, 1.0f, 0.0f, -1.0f}),
+            "ctc_greedy_decoder_multiple_sequences"),
+    };
+    return params;
+}
+
+std::vector<CTCGreedyDecoderParams> generateCombinedParams() {
+    const std::vector<std::vector<CTCGreedyDecoderParams>> generatedParams {
+        generateParams<element::Type_t::bf16>(),
```

Comment:
In the operation semantic I don't see any limitations for integer or other types. Could we extend the set of data types?

Reply:
In https://docs.openvino.ai/latest/openvino_docs_ops_sequence_CTCGreedyDecoder_1.html, input and output types are float point.